tests/app/billing/test_billing.py

```diff
@@ -395,5 +394,58 @@ def test_post_free_sms_fragment_limit_current_year(client, sample_service):
 
     assert response.status_code == 201
     assert response_get.status_code == 200
-    assert json_resp['data']['financial_year_start'] == get_current_financial_year_start_year()
-    assert json_resp['data']['free_sms_fragment_limit'] == 7777
+    assert json_resp['financial_year_start'] == get_current_financial_year_start_year()
+    assert json_resp['free_sms_fragment_limit'] == 7777
+
+
+def test_get_free_sms_fragment_limit_past_year_get_db_oldest_record(client, sample_service):
+    response = client.get(
+        'service/{}/billing/free-sms-fragment-limit'.format(sample_service.id),
+        headers=[('Content-Type', 'application/json'), create_authorization_header()])
+    sms_list = json.loads(response.get_data(as_text=True))
+    past_year = sms_list[0]['financial_year_start'] - 1
+
+    # create an older entry than the one created in sample service has an entry already
+    annual_billing = {'financial_year_start': past_year, 'free_sms_fragment_limit': 1000}
+    response = client.post('service/{}/billing/free-sms-fragment-limit'.format(sample_service.id),
+                           data=json.dumps(annual_billing),
+                           headers=[('Content-Type', 'application/json'), create_authorization_header()])
+
+    response = client.get(
+        'service/{}/billing/free-sms-fragment-limit?financial_year_start={}'.format(sample_service.id, past_year - 2),
+        headers=[('Content-Type', 'application/json'), create_authorization_header()])
+    json_resp = json.loads(response.get_data(as_text=True))
+
+    assert response.status_code == 200
+    assert json_resp['free_sms_fragment_limit'] == 1000
+
+
+def test_get_free_sms_fragment_limit_future_year_get_and_create_db_newest_record(client, sample_service):
+
+    response = client.get(
+        'service/{}/billing/free-sms-fragment-limit'.format(sample_service.id),
+        headers=[('Content-Type', 'application/json'), create_authorization_header()])
+    sms_list = json.loads(response.get_data(as_text=True))
+    next_year = sms_list[-1]['financial_year_start'] + 1
+
+    # create an older entry than the one created in sample service has an entry already
+    annual_billing = {'financial_year_start': next_year, 'free_sms_fragment_limit': 1000}
+    response = client.post('service/{}/billing/free-sms-fragment-limit'.format(sample_service.id),
+                           data=json.dumps(annual_billing),
+                           headers=[('Content-Type', 'application/json'), create_authorization_header()])
+    # request one year ahead that record doesn't exist yet
+    response = client.get(
+        'service/{}/billing/free-sms-fragment-limit?financial_year_start={}'.format(sample_service.id, next_year + 2),
+        headers=[('Content-Type', 'application/json'), create_authorization_header()])
+    json_resp = json.loads(response.get_data(as_text=True))
+
+    # get a list of request and see if the newe
+    response = client.get(
```

Comment:
If you are writing a test to make sure that a new record is created for a year it's probably better to use a DAO function / data model query to check that the data is as expected.